util/ipgen/lib.py

```diff
@@ -95,7 +98,13 @@ def _parse_template_parameter(where: str, raw: object) -> TemplateParameter:
     else:
         assert False, f"Unknown parameter type found: {param_type!r}"
 
-    return TemplateParameter(name, desc, param_type, default)
+    # DT generator dictionary is not checked here, it is only forwarded to dtgen
```

Comment:
I really like the careful comments!

I think this has to be a dictionary, at least. Maybe it's worth adding that check here?